src/extension.ts

```diff
@@ -242,8 +242,15 @@ export async function activate(context: vscode.ExtensionContext) {
         const coverageOptions = getCoverageOptions();
         covRenderer = new CoverageRenderer(workspaceRoot, coverageOptions);
       }
+      const projectName = await getProjectName(workspaceRoot.fsPath);
+      const projectElfFile = `${path.join(
+        workspaceRoot.fsPath,
+        "build",
+        projectName
+      )}.elf`;
       const debugAdapterConfig = {
         currentWorkspace: workspaceRoot,
+        elfFile: projectElfFile,
```

Comment:
What if elf file doesn't exists ?

Reply:
I think the debug adapter does require the elfFile to work. Isn't that the case @an-dr ?

Reply:
This is true, but what will happen if user launch debug without build ...ideally we should trigger build but showing an alert works as well.

Reply:
We cant work without elf, bacause in the file we have connections between sources and binary code. Personally me would be glad to have a notification (There is no elf-file or smth) with a button "Build now" in that situation

Reply:
Note that if we rebuild the ELF file, it may no longer match the program running on the target.
So if the ELF file doesn't exist, and we rebuild it, we should also upload the program again, before starting to debug.

One way to check this is to compare the SHA256(elf file) with the value of SHA256 printed by the panic handler, before running the GDB Stub. This would require changes on the IDF Monitor side, though...

Reply:
I added a simple elf exists check that shows an option to either build and flash or don't. The debug session will not start automatically after flash (i.e. need to press F5 again after flashing).